### ISSUE TITLE
5.9.0 undocumented target change to ES6 was breaking change for me

### ISSUE THREAD

**Issue by 1jerry** [szokodiakos/typegoose#368]
Without changing my code, my deployments stopped working.  I tracked it down to an upgrade to typegoose, which my deployment process installed (with npm install --production) which I had not yet updated on the dev and QA machines.  typegoose 5.8.1 (and below) has been working fine for quite a while.  When it updated to 5.9.0 my code would no longer run, producing the following error.  (easily duplicated on my dev machine by updating typegoose to 5.9.0 there also.  _nothing else_ but updating typegoose.). 

## Error:
```
        return _super !== null && _super.apply(this, arguments) || this;
                                         ^

TypeError: Class constructor Typegoose cannot be invoked without 'new'
```
## Versions

- NodeJS: 8,10,11
- Typegoose(NPM): 5.9.0
- mongoose: 5.6.7

## Code Example

```ts
export class callLeg extends Typegoose {
    @prop()   id: string
...
```
transpiles into:
```js
var callLeg = /** @class */ (function (_super) {
    __extends(callLeg, _super);
    function callLeg() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        typegoose_1.prop(),
        __metadata("design:type", String)
    ], callLeg.prototype, "id", void 0);
...
```

## Do you know *why* it happens?

Something in 5.9.0 is causing this error, on a line of code that should be very straightforward.

**Comment by hasezoey** [comment]
first notes:
- typegoose 5.9.0 "requires" mongoose 5.6.7+
- could you post the tsconfig / say what your target for your project is? (es5?)

**Comment by 1jerry** [comment]
I _did_ use Mongoose 5.6.7.  It's just I was experimenting with many different things, and I wrote the wrong number, which I have corrected.  The deploy process just does a _npm install_, so it gets the latest version available of everything.

Here is my tsconfig:
```
{
    "compilerOptions": {
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "esModuleInterop": true,
        "outDir": "./dist",
        "allowJs": true,
        "lib": [
            "dom",
            "es2017"
        ],
        "sourceMap": true,
        "baseUrl": ".",
        "paths": { "*": ["types/*"] },
        "target": "es5"
    },
    "include": [
        "./src/**/*"
    ]
}

```

**Comment by hasezoey** [comment]
> "dom",

are you using typegoose in the browser?

> "target": "es5"

typegoose (npm package) compiles to ES6, if you want to use ES5, you must use typegoose as a gitsubmodule, and reference it directly (`import * as typegoose from './typegoose/src/typegoose'`)

**Comment by 1jerry** [comment]
I've always wondered about "dom" being there, but it never cause any harm so I didn't remove it.

I have used the same "target": "es5" ever since using typegoose and everything worked fine.  There is no mention of ES5 vs ES6 in the documentation.  So something changed that is a breaking change for me, and anyone else that has that setting.

I will adjust my tsconfig and let you know how it goes.

Thanks for the quick reply.

**Comment by hasezoey** [comment]
sorry, thought it was in the release notes, it should have been, but here is the commit that changed it ea1c0ec6bbebdd6b27ba4580ec39c1d3535b2084

**Comment by 1jerry** [comment]
I got version 5.9.0 to work now.
For future reference, I also had to add the line "module": "commonjs", in tsconfig because changing the target to ES6 will also change the default for the "module" setting.  If I _just_ changed the "target" setting, 4 of my npm packages could not be found (including typegoose) when I went to compile with TypeScript.  Why?  I haven't a clue.  But it seemed to be a TypeScript issue, not a typegoose one.
